src/main/java/Duke.java

```diff
@@ -1,10 +1,87 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner in = new Scanner(System.in);
+        ArrayList<Task> tasks = new ArrayList<Task>();
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        int numOfTasks = 0;
+        // parses the user input
+        while (true) {
+            String input = in.nextLine();
+            if (input.equals("bye")) {
+                break;
+            } else if (input.equals("list")) {
+                list(numOfTasks, tasks);
+            } else if (input.matches("done(.*)")) {
+                markDone(input, tasks);
+            } else if (input.matches("todo.*")){
+                addTodo(input, tasks, numOfTasks);
+                numOfTasks++;
+            } else if (input.matches("deadline.*")) {
+                addDeadline(input, tasks, numOfTasks);
+                numOfTasks++;
```

Comment:
You may also want to put the numOfTasks as a static variable for Task, so that you do not need to put it inside the parameter every time you want to add a Task or its subclasses

Reply:
Thanks for the suggestion